lib/ansible/modules/storage/netapp/na_ontap_volume.py

```diff
@@ -0,0 +1,614 @@
+            volume_export_attributes = volume_attributes.get_child_by_name(
+                'volume-export-attributes')
+            policy = volume_export_attributes.get_child_content('policy')
+            space_guarantee = volume_space_attributes.get_child_content(
+                'space-guarantee')
+            percent_snapshot_space = volume_space_attributes.get_child_by_name(
+                'percentage-snapshot-reserve')
+
+            is_online = None
+            if current_state == "online":
+                is_online = True
+            elif current_state == "offline":
+                is_online = False
+            return_value = {
+                'name': self.name,
+                'size': current_size,
+                'is_online': is_online,
+                'aggregate_name': aggregate_name,
+                'policy': policy,
+                'space_guarantee': space_guarantee,
+                'percent_snapshot_space': percent_snapshot_space,
+                'type': volume_type,
+                'junction_path': junction_path
+
+            }
+
+        return return_value
+
+    def create_volume(self):
+        '''Create ONTAP volume'''
+        if self.aggregate_name is None:
+            self.module.fail_json(msg='Error provisioning volume %s: \
+                                  aggregate_name is required'
+                                  % self.name)
+        options = {'volume': self.name,
+                   'containing-aggr-name': self.aggregate_name,
+                   'size': str(self.size)}
+        if self.percent_snapshot_space is not None:
+            options['percentage-snapshot-reserve'] = self.percent_snapshot_space
+        if self.type is not None:
+            options['volume-type'] = self.type
+        if self.policy is not None:
+            options['export-policy'] = self.policy
+        if self.junction_path is not None:
+            options['junction-path'] = self.junction_path
+        if self.space_guarantee is not None:
+            options['space-reserve'] = self.space_guarantee
+        if self.volume_security_style is not None:
+            options['volume-security-style'] = self.volume_security_style
+
+        volume_create = netapp_utils.zapi\
+            .NaElement.create_node_with_children(
+                'volume-create', **options)
+        try:
+            self.server.invoke_successfully(volume_create,
+                                            enable_tunneling=True)
+            self.ems_log_event("create")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error provisioning volume %s \
+                                  of size %s: %s'
+                                  % (self.name, self.size, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def delete_volume(self):
+        '''Delete ONTAP volume'''
+        if self.is_infinite:
```

Comment:
docs should probably specify that deleting an infinite volume is not synchronous